packages/server/safetyPatchRes.ts

```diff
@@ -69,17 +81,17 @@ const safetyPatchRes = (res: HttpResponse) => {
     if (res.done) {
       console.warn(`uWS: Called writeHeader after done`)
     }
-    if (res.done || res.aborted) return res
-    return res._writeHeader(key, value)
+    res.headers.push([key, value])
```

Comment:
This is the crux of it. uWS gets mad when we call write methods outside of a cork, so we capture those writes & save them for later when we're ready to cork.